### ISSUE TITLE
Proper way to get path to the config file that was parsed

### ISSUE THREAD

**Issue by pwm1234** [CLIUtils/CLI11#236]
What is the right way to get the path to the config file that was parsed? For debugging purposes, I want to log the path to the config file that was loaded. Since the default path could have been overridden during argument parsing, I need the Option returned by the ``app->get_config_ptr()`` call to tell me. At least this is how I have done it:

    CLI::Option* config = app.get_config_ptr();
    std::vector<std::string> results = config->results();
    std::string config_file = config->get_defaultval();
    if (results.size() > 0)
    {
        config_file = results[0];
    }

Is there a better way? If not, should there be a better way? If this is not obvious, should this be added to the documentation?

**Comment by phlptp** [comment]
After #235 is merged you should be able to do

```
config_file = app.get_config_ptr()->as<std::string>();
```
or 
```
std::string config_file;
app.get_config_ptr()->results(config_file);
```

until then I think what you did might be the only way.  though @henryiii  might know something else.

**Comment by pwm1234** [comment]
Wow--you did this and more before I even asked for it!!! Thank you.

By the way, what is the process (and delay) for updating CLI11 in vcpkg? I have kind of settled upon vcpkg for getting almost all of my 3rd party dependencies, so I would like to understand how and when CLI11 changes make it into the vcpkg head. (Should I make this question a separate issue?)

**Comment by pwm1234** [comment]
I just discovered another way to get the config value using ``check`` with a lambda function. This seems better and more concise than my first attempt:

```cpp
auto config = app.set_config("--config", config_val);
config->check([&](const std::string &config_arg) { 
    config_val = config_arg;
    return config_arg;
});
```

This is my first attempt using a custom validator. While your PR changes provide exactly what I need, this (ab)use of ``check`` seems to provide an alternative to poking into the results.

**Comment by phlptp** [comment]
I needed it for some options where I don't have any place to store the result until after the parsing, so this made it much easier to deal with that in a nice way, and make converting from boost::program_option somewhat more straightforward.

**Comment by phlptp** [comment]
I think in your check function you probably need to return an empty string otherwise it will throw an error of some kind since the return string usually goes into an error message about the validation failing.

**Comment by phlptp** [comment]
Or you could put that into the translate instead of the check

**Comment by henryiii** [comment]
Yes, @phlptp is correct. You'll want to return an empty string for "no error". However, what about running each? That sounds pretty close to what you want.

```cpp
auto config = app.set_config("--config", config_val);
config->each([&config_val](const std::string &config_arg) { 
    config_val = config_arg;
});
```

**Comment by pwm1234** [comment]
Yes, ``each`` sounds even better. It was, however, good for me to learn how to use a validator. 
Thanks!
